status/store.go

```diff
@@ -0,0 +1,168 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+)
+
+type Store struct {
+	eventRecorder      record.EventRecorder
+	observedConditions sync.Map // map[reconcile.Request]ConditionSet
+	terminatingObjects sync.Map // map[reconcile.Request]DeletionTimestamp
+}
+
+func NewStore(eventRecorder record.EventRecorder) Store {
```

Comment:
Do we need this store implementation or can we wrap the standard status controller with a client.Object controller and just reuse its reconcile metric so we don't have to copy all of the logic and data store?

Reply:
I have updated the PR to taken the changes